packages/babel-plugin-transform-template-literals/src/index.js

```diff
@@ -83,19 +82,27 @@ export default declare((api, options) => {
           callExpressionInput.push(t.arrayExpression(raws));
         }
 
-        const init = t.callExpression(helperId, callExpressionInput);
-        annotateAsPure(init);
-        init._compact = true;
         scope.push({
           id: templateObject,
-          init,
           // This ensures that we don't fail if not using function expression helpers
           _blockHoist: 1.9,
         });
 
+        const lazyLoad = t.logicalExpression(
+          "||",
+          t.cloneNode(templateObject),
+          t.callExpression(helperId, callExpressionInput),
+        );
+
+        const conditionalAssignment = t.assignmentExpression(
+          "=",
+          t.cloneNode(templateObject),
+          lazyLoad,
+        );
+
         path.replaceWith(
           t.callExpression(node.tag, [
-            t.cloneNode(templateObject),
+            t.parenthesizedExpression(conditionalAssignment),
```

Comment:
Why is this `ParenthesizedExpression` needed?

Reply:
Actually, it is not.:D  I didn't realize that this is an assignment expression that returns a value. I will remove it. Thanks for the feedback. 👍